src/js/components/inspectors/Scale.tsx

```diff
@@ -18,29 +21,103 @@ interface StateProps {
   scale: ScaleRecord;
 }
 
+interface DispatchProps {
+  updateScaleProperty: (scaleId: number, property: string, value: any) => void;
+}
+
 
 function mapStateToProps(state: State, ownProps: OwnProps): StateProps {
   return {
     scale: getInVis(state, 'scales.' + ownProps.primId)
   };
 }
 
-class BaseScaleInspector extends React.Component<OwnProps & StateProps> {
+function mapDispatchToProps(dispatch: Dispatch): DispatchProps {
+  return {
+    updateScaleProperty: function (scaleId, property, value) {
+      dispatch(updateScaleProperty({ property, value }, scaleId));
+    }
+  };
+}
+
+class BaseScaleInspector extends React.Component<OwnProps & StateProps & DispatchProps> {
+
+  public handleChange(evt) {
+    const scaleId = this.props.primId;
+    const target = evt.target;
+    const property = target.name;
+
+    let value = (target.type === 'checkbox') ? target.checked : target.value;
+    // Parse number or keep string around.
+    value = value === '' || isNaN(+value) ? value : +value;
+    this.props.updateScaleProperty(scaleId, property, value);
+  };
+
+  public handleRangeColor(evt) {
```

Comment:
not sure if there is a way to handle this in the other `handleChange()` function but need to pass an object for the color scheme value

Reply:
gotcha. moved it to handlechange. lmk if it looks ok.